packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -142,6 +142,7 @@ export class AuthUtil {
 
             if (!this.isConnected()) {
                 await this.regionProfileManager.invalidateProfile(this.regionProfileManager.activeRegionProfile?.arn)
+                await this.regionProfileManager.resetCache()
```

Comment:
should `invalidateProfile` do this internally?

Reply:
this is kinda redundant i think, and also its name is confusing for sure (should've named it `releaseLock`), but i will remove it.

Reply:
`invalidateProfile` doesn't need this i think as the source of truth of profile `invalidateProfile`  is also from this resourceCache. Originally i added this line is kinda for debugging purpose and to avoid a dead lock scenario when the code was problematic. However i think releaseLock will be called
1. when `getResource` goes to pull real response and executes successfully
2. when `getResource` goes to pull real response and executes exceptionally
3. when `getResource` goes to return cached value
4. if one process/ide instance waits to long (current it's set to 15s), it will go either (1) or (2) and eventually release the lock again

so I think it's redundant

Reply:
just reminds me that  i need to add one `clearCache` onConnectionChanged.